src/permissionGroups/utils.ts

```diff
@@ -113,26 +113,100 @@ export const arePermissionsExceeded = (
  */
-export const getPermissionGroupAccessibleChannels = (
+export const mapAccessibleChannelsToChoice = (
   permissionGroup: PermissionGroupWithContextDetailsFragment,
-  allChannelsLength: number,
-): MultiAutocompleteChoiceType[] => {
-  // We don't want show all channels to user that has no restricted access to channels
-  // User will be able to select channels manually
+  isUserAbleToEdit?: boolean,
+): MultiAutocompleteChoiceType[] =>
+  permissionGroup?.accessibleChannels.map(
+    channel =>
+      ({
+        label: channel.name,
+        value: channel.id,
+        disabled: isUserAbleToEdit !== undefined ? !isUserAbleToEdit : false,
+      } as unknown as MultiAutocompleteChoiceType),
+  ) ?? [];
+
+/**
+ * Calcualte if restricted access to channels should be enabled.
+ */
+export const checkIfPermissionGroupHasRestrictedChannels = (
+  hasUserRestrictedChannels: boolean,
+  selectedChannels: unknown[],
+  allChannels: unknown[],
+) => {
+  // When user has restricted access to channels we know that group has restricted access to channels.
+  if (hasUserRestrictedChannels) {
+    return true;
+  }
+
+  // When user has full access and selected less channel group has restricted access to channels.
   if (
-    permissionGroup?.accessibleChannels?.length === allChannelsLength &&
-    !permissionGroup?.restrictedAccessToChannels
+    !hasUserRestrictedChannels &&
+    selectedChannels.length !== allChannels.length
   ) {
-    return [];
+    return true;
   }
 
-  return (
-    permissionGroup?.accessibleChannels.map(
-      channel =>
-        ({
-          label: channel.name,
-          value: channel.id,
-        } as unknown as MultiAutocompleteChoiceType),
-    ) ?? []
+  return false;
+};
+
+/**
+ * User is eligible to edit channels when he has access to all channels in permission group.
+ */
+export const checkIfUserIsEligibleToEditChannels = (
+  user: UserContext["user"],
+  permissionGroupAccessibleChannels: ChannelFragment[],
+) => {
+  const userChannels = getUserAccessibleChannels(user).map(c => c.id);
+
+  return permissionGroupAccessibleChannels.every(permChan =>
+    userChannels.includes(permChan.id),
   );
 };
+
+/**
+ * Get channels options for select field.
+ */
+export const getChannelsOptions = (
+  availableChannels: ChannelFragment[],
+  user?: UserContext["user"],
+): ChannelFragment[] => {
+  if (!user) {
+    return availableChannels;
+  }
+
+  if (
+    "restrictedAccessToChannels" in user &&
+    user.restrictedAccessToChannels === false
+  ) {
+    return availableChannels;
+  }
+
+  if ("accessibleChannels" in user) {
+    return user.accessibleChannels;
+  }
+
+  return availableChannels;
+};
+
+/**
+ * Check if user has restricted access to channels.
+ */
+export const checkIfUserHasRestictedChannels = (user?: UserContext["user"]) => {
+  if (user && "restrictedAccessToChannels" in user) {
+    return user.restrictedAccessToChannels;
+  }
+
+  return false;
```

Comment:
**suggestion**: won't this work this way?
```suggestion
  return user?.restrictedAccessToChannels ?? false
```

Reply:
Unfortunately won't work this way because `restrictedAccessToChannels` is field that only exists under feature flag